src/hotspot/os/linux/os_linux.cpp

```diff
@@ -2362,6 +2371,35 @@ bool os::Linux::query_process_memory_info(os::Linux::meminfo_t* info) {
   return false;
 }
 
+// Accurate memory information need Linux 4.14 or newer
+bool os::Linux::query_accurate_process_memory_info(os::Linux::accurate_meminfo_t* info) {
+  FILE* f = os::fopen("/proc/self/smaps_rollup", "r");
+  const int num_values = sizeof(os::Linux::accurate_meminfo_t) / sizeof(size_t);
+  int num_found = 0;
+  char buf[256];
+  info->rss = info->pss = info->pssdirty = info->pssanon =
+      info->pssfile = info->pssshmem = info->swap = info->swappss = -1;
+  if (f != nullptr) {
```

Comment:
You could consider inverting this check, and early-return `false` if `f==nullptr`. That would spare one indentation level